kuser/kinit.c

```diff
@@ -1331,6 +1337,16 @@ main(int argc, char **argv)
 	    krb5_err(context, 1, ret, "krb5_make_principal");
 	krb5_principal_set_type(context, principal, KRB5_NT_WELLKNOWN);
 	anonymous_pkinit = TRUE;
+    } else if (anonymous_flag && historical_anon_pkinit) {
+        char *realm = argc == 0 ? get_default_realm(context) :
```

Comment:
Potential memory leak, `get_default_realmI()` returns allocated memory.

Reply:
This code runs exactly once in `main()`. I am not inclined to bother...